### DISCUSSION TITLE
action is not trigged by file change

### DISCUSSION THREAD

**Discussion by iivvaall** [community/community#26992] [Actions]
<p>in the .github/workflows/runner.yaml</p>
<pre><code>on:
  push:
    branches:
      - master
    paths:
      - 'runner/**'
      - '.github/workflows/runner.yaml'
  pull_request:
    branches:
      - master
    paths:
      - 'runner/**'
      - '.github/workflows/runner.yaml'
</code></pre>
<p>I change the .github/workflows/runner.yaml in the PR and action is not triggered<br>
Without “paths:” the workflow runs but I do not wont it to be triggered on every PR.<br>
Could you please tell me whats is wrong?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/iivvaall">@iivvaall</a> ,</p>
<blockquote>
<p>I change the .github/workflows/runner.yaml in the PR and action is not triggered</p>
</blockquote>
<p>From which branch the PR is merging? And into which branch the PR is merging?<br>
In which branch did you change the " <strong>.github/workflows/runner.yaml</strong>" file?</p>
<p>According to the configurations of the triggers in your workflow, the workflow can be triggered in the following situations:</p>
<p><strong>1)</strong> Directly push commits to the <strong>master</strong> branch, and at least one modified file in the commits can match the patterns (’ <strong>runner/**</strong>’ or ’ <strong>.github/workflows/runner.yaml</strong>’) of the <strong>paths filter</strong>. In this situation, the workflow YAML file must have been existing in the <strong>master</strong> branch, and the workflow is triggered by the <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#push-event-push" rel="nofollow noopener"><strong>Push event</strong></a>.</p>
<p><strong>2)</strong> After pushing some commits in a branch (not the <strong>master</strong> branch, suppose it’s named " <strong>develop</strong>"), when creating and opening a PR to merge the commits from the <strong>develop</strong> branch into the <strong>master</strong> branch, at least one modified file in the commits can match the patterns (’ <strong>runner/**</strong>’ or ’ <strong>.github/workflows/runner.yaml</strong>’) of the <strong>paths filter</strong>. In this situation, the workflow YAML file must have been existing in the <strong>develop</strong> branch, and the workflow is triggered by the <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener"><strong>Pull Request event</strong></a> ( <strong>Activity type: opened</strong> ).</p>
<p><strong>3)</strong> After opening the PR (the PR is " <strong>Open</strong>" status), when you push some commits to the <strong>develop</strong> branch, at least one modified file in the commits can match the patterns (’ <strong>runner/**</strong>’ or ’ <strong>.github/workflows/runner.yaml</strong>’) of the <strong>paths filter</strong>. In this situation, the workflow YAML file must be existing in the <strong>develop</strong> branch, and the workflow is triggered by the <strong>Pull Request event</strong> ( <strong>Activity type: synchronize</strong> ).</p>
<p><strong>4)</strong> If you close the PR without merged (the PR is " <strong>Closed</strong>" status, not " <strong>Merged</strong>" status), then <strong>reopen</strong> this PR. In this situation, the workflow YAML file must be existing in the <strong>develop</strong> branch, and the workflow is triggered by the <strong>Pull Request event</strong> ( <strong>Activity type: reopened</strong> ).</p>
<p>So, please confirm whether it satisfies any of the above situations, when you change the " <strong>.github/workflows/runner.yaml</strong>" file.</p>
<p>In addition, if you run a workflow to modify and push files with using the <strong>GITHUB_TOKEN</strong> to authenticate, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs (see <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener">here</a>).</p>

**Comment by iivvaall** [comment]
<blockquote>
<p><a class="mention" href="/brightran">@brightran</a> wrote:</p>
</blockquote>
<blockquote>
<p><a class="mention" href="/iivvaall">@iivvaall</a> ,</p>
</blockquote>
<blockquote>
<blockquote>
<p>I change the .github/workflows/runner.yaml in the PR and action is not triggered</p>
</blockquote>
</blockquote>
<blockquote>
<p>From which branch the PR is merging? And into which branch the PR is merging?<br>
In which branch did you change the " <strong>.github/workflows/runner.yaml</strong>" file?</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p><strong>3)</strong> After opening the PR (the PR is " <strong>Open</strong>" status), when you push some commits to the <strong>develop</strong> branch, at least one modified file in the commits can match the patterns (’ <strong>runner/**</strong>’ or ’ <strong>.github/workflows/runner.yaml</strong>’) of the <strong>paths filter</strong>. In this situation, the workflow YAML file must be existing in the <strong>develop</strong> branch, and the workflow is triggered by the <strong>Pull Request event</strong> ( <strong>Activity type: synchronize</strong> ).</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>Situation number 3. I am just starting using the github actions.</p>
<p>So I created new branch, added ./.github/workflows/runner.yaml, created PR and started pushing commits.</p>
<p>I tried to reproduce this situation in public repository (<a href="https://github.com/iivvaall/actions_test" rel="nofollow noopener">https://github.com/iivvaall/actions_test</a>) and every thing just works. </p>
<p>I double checked the path. They are OK.</p>
<p>What else cound influent the build? Any idea?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/iivvaall">@iivvaall</a> ,</p>
<p>I tested on my private and public repositories with the same workflow configurations as you posed, and I also did not reproduce the problem you reported, everything works fine as expected. This may be a temporary problem.</p>
<p>Please try again in your repository to see if the problem has gone on your side.</p>
<p>If the problem still exists, please feel free to tell me, and I will help you report this question to the appropriate engineering team for further investigation and evaluation.</p>
<p>◆ If your repository which has the problem is private, please share the repository’s name (e.g. <strong>myOrg/myRepo</strong> ) with us.</p>
<p>◆ If your repository which has the problem is public, please share the repository’s link with us.</p>

**Comment by iivvaall** [comment]
<p>I have just tested the workflow. It runs. It seems to be a temporary problem. Thank you!</p>
